pkg/service/connect_service.go

```diff
@@ -144,25 +144,29 @@ func (s *ConnectService) EventStream(
 	req *connect.Request[emptypb.Empty],
 	stream *connect.ServerStream[schemaV1.EventStreamResponse],
 ) error {
-	s.eventingConfiguration.subs[req] = make(chan Notification, 1)
+	requestNotificationChan := make(chan Notification, 1)
+	s.eventingConfiguration.mu.Lock()
+	s.eventingConfiguration.subs[req] = requestNotificationChan
+	s.eventingConfiguration.mu.Unlock()
 	defer func() {
 		s.eventingConfiguration.mu.Lock()
 		delete(s.eventingConfiguration.subs, req)
 		s.eventingConfiguration.mu.Unlock()
 	}()
-	s.eventingConfiguration.subs[req] <- Notification{
+	requestNotificationChan <- Notification{
 		Type: ProviderReady,
 	}
+	ticker := time.NewTicker(20 * time.Second)
 	for {
 		select {
-		case <-time.After(20 * time.Second):
+		case <-ticker.C:
```

Comment:
I assumed that this is meant to occur every 20 seconds but will remove if I'm wrong. @james-milligan can hopefully shine some light